sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/ForeachWriterTable.scala

```diff
@@ -141,17 +143,33 @@ class ForeachDataWriter[T](
       writer.process(rowConverter(record))
     } catch {
       case t: Throwable =>
-        writer.close(t)
+        closeWriter(t)
         throw t
     }
   }
 
   override def commit(): WriterCommitMessage = {
-    writer.close(null)
+    closeWriter(null)
     ForeachWriterCommitMessage
   }
 
-  override def abort(): Unit = {}
+  override def abort(): Unit = {
+    closeWriter(new RuntimeException("Foreach writer has been aborted"))
```

Comment:
It's better to use `SparkException`. How about `new SparkException("Foreach writer has been aborted due to a task failure")`?

Reply:
fine by me :)